monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -120,16 +129,26 @@ def _send_exploit_rpc_login_requests(self, rpc_con) -> bool:
         return False
 
     def try_exploit_attempt(self, rpc_con) -> Optional[object]:
+        error_message = ""
         try:
             exploit_attempt_result = self.attempt_exploit(rpc_con)
             return exploit_attempt_result
-        except nrpc.DCERPCSessionError as e:
+        except nrpc.DCERPCSessionError as err:
             # Failure should be due to a STATUS_ACCESS_DENIED error.
             # Otherwise, the attack is probably not working.
-            if e.get_error_code() != self.ERROR_CODE_ACCESS_DENIED:
-                logger.info(f"Unexpected error code from DC: {e.get_error_code()}")
-        except BaseException as e:
-            logger.info(f"Unexpected error: {e}")
+            if err.get_error_code() != self.ERROR_CODE_ACCESS_DENIED:
+                error_message = f"Unexpected error code from DC: {err.get_error_code()}"
+                logger.info(error_message)
+        except BaseException as err:
```

Comment:
Don't use `BaseException`. It will catch system errors and keyboard interrupts. It's a one-way ticket to a hanging process